hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/ColumnStatsIndexSupport.scala

```diff
@@ -116,8 +116,8 @@ class ColumnStatsIndexSupport(spark: SparkSession,
    * w/in the Metadata Table
    */
   def isIndexAvailable: Boolean = {
-    checkState(metadataConfig.isEnabled, "Metadata Table support has to be enabled")
-    metaClient.getTableConfig.getMetadataPartitions.contains(HoodieTableMetadataUtil.PARTITION_NAME_COLUMN_STATS)
+    metadataConfig.isEnabled &&
```

Comment:
avoid interrupt index select in 
```
if (isDataSkippingEnabled) {
      for(indexSupport: SparkBaseIndexSupport <- indicesSupport) {
        if (indexSupport.isIndexAvailable) {
          val prunedFileNames = indexSupport.computeCandidateFileNames(this, queryFilters, queryReferencedColumns,
            prunedPartitionsAndFileSlices, shouldPushDownFilesFilter)
          if (prunedFileNames.nonEmpty) {
            return Try(prunedFileNames)
          }
        }
      }
    }
```

Reply:
makes sense, and another improvement that I can think of is the semantics of empty `prunedFileNames`, there is case the files are actually empty in rare cases if all the files are filtered out, and we should return early instead of moving to another index.

The best way we can do is making it orthoganal, i.e. to extends the pruned files from preceding index pruning.